src/coreclr/vm/jitinterface.cpp

```diff
@@ -7890,6 +7890,25 @@ CorInfoInline CEEInfo::canInline (CORINFO_METHOD_HANDLE hCaller,
         }
     }
 
+    // If the root level caller and callee modules do not have the same runtime
+    // wrapped exception behavior, and the callee has EH, we cannot inline.
+    _ASSERTE(!pCallee->IsDynamicMethod());
+    {
+        COR_ILMETHOD_DECODER header(pCallee->GetILHeader(), pCallee->GetMDImport(), NULL);
+        if (header.EHCount() > 0)
+        {
+            Module* pCalleeModule = pCallee->GetModule();
+            Module* pRootModule = pOrigCaller->GetModule();
+
+            if (pRootModule->IsRuntimeWrapExceptions() != pCalleeModule->IsRuntimeWrapExceptions())
```

Comment:
I think this needs to compare method being compiled and the inlinee to deal with cases like this one correctly:

methodFromModuleA -> methodFromModuleB -> methodFromModuleBWithEH

Reply:
Ah, it does that already. pOrigCaller is m_pMethodBeingCompiled.